mod/01_core/format/html_format.rb

```diff
@@ -216,8 +265,8 @@ def unsupported_view view
     end
 
     def final_link href, opts={}
-      text = opts[:text] || href
-      %{<a class="#{opts[:class]}" href="#{href}">#{text}</a>}
+      text = (opts.delete(:text) || href).dup
+      content_tag :a, text, opts.merge(:href=>href)
```

Comment:
You'll note that I changed those back and compensated by changing the expected result to an encoded &.

Also added test case for text rendering of links which should not change a regular url.  If the recognized url is missing something (like http://, a bare host+path), then the url is different from the parsed text, so it does the [] thing.